src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] list;
+    private int size;
+
+    public ArrayList() {
+        list = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == list.length) {
+            grow();
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index + " is out of bounds!");
+        }
+        if (size == list.length) {
+            grow();
+        }
+        if (size > index) {
+            System.arraycopy(list, index, list, index + 1, size);
+        }
+        list[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (isIndexInInvalidRange(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index + " is out of bounds!");
+        }
+        return list[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (isIndexInInvalidRange(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index + " is out of bounds!");
+        }
+        list[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (isIndexInInvalidRange(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index + " is out of bounds!");
+        }
+        T removedObject = list[index];
+        removeObject(index);
+        return removedObject;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = findIndexByElement(element);
+        if (index == -1) {
+            throw new NoSuchElementException("The element was not found!");
+        } else {
+            removeObject(index);
+            return element;
+        }
```

Comment:
else statement is redundant here
```suggestion
        }
        removeObject(index);
        return element;
```